cobigen-templates/templates-oasp4j/src/main/java/com/devonfw/cobigen/templates/oasp4j/utils/OaspUtil.java

```diff
@@ -433,17 +433,17 @@ public String getListArgumentType(Map<String, Object> field, Class<?> pojoClass)
 
     /**
      * Converts all occurrences of OASP Entities types in the given 'field' simple type (possibly generic) to
-     * Longs
+     * Entities
      *
      * @param field
      *            the field
-     * @return the field type as String. If field type contains 'Entity' the result is Long
+     * @return the field type as String. If field type contains 'Entity' the result is IdRef<E>
      */
     public String getSimpleEntityTypeAsLongReference(Map<String, Object> field) {
 
         String fieldType = (String) field.get(Field.TYPE.toString());
         if (fieldType.contains("Entity")) {
-            fieldType = fieldType.replaceAll("[^<>]+Entity", "Long");
+            fieldType = fieldType.replaceAll("[^<>]+Entity", "IdRef<E>");
```

Comment:
This would return `IdRef<E>` which would result in a compilation error as `E` is not a valid type.

Most probably it should be `fieldType.replaceAll("[^<>]+Entity", "IdRef<" + field.type + ">");`